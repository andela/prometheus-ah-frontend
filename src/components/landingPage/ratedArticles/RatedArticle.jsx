import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const RatedArticle = ({
  title, body, author, date, active
}) => {
  const formatedDate = moment(date).format('MMM DD, YYYY');
  const truncatedString = body.slice(0, 120);
  let classname = 'carousel-item slide';
  if (active === 0) {
    classname = 'carousel-item active slide';
  }
  return (
    <div className={classname}>
      <h2 className="carousel-article-title">{title}</h2>
      <p className="carousel-author-name">{`${author} - ${formatedDate}`}</p>
      <p>{`${truncatedString}...`}</p>
      <button type="button" className="btn read-more">Read more</button>
    </div>
  );
};

RatedArticle.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  author: PropTypes.string,
  date: PropTypes.string,
  active: PropTypes.number
};

export default RatedArticle;
