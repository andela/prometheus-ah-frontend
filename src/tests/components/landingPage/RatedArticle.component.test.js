import React from 'react';
import { shallow } from 'enzyme';
import RatedArticle from '../../../components/landingPage/ratedArticles/RatedArticle';
import mockData from '../../__mocks__/mockData';

let wrapper;

describe('RecentPost component', () => {
  beforeEach(() => {
    const {
      title, body, author, date, active
    } = mockData.postProps;
    const props = {
      title, body, author, date, active
    };

    wrapper = shallow(<RatedArticle {...props} />);
  });

  test('should render a snapshot of the RecentPost component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
