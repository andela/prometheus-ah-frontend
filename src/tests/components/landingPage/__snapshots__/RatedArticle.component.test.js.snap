// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentPost component should render a snapshot of the RecentPost component 1`] = `
<div
  className="carousel-item active slide"
>
  <h2
    className="carousel-article-title"
  >
    My Broken Mother, My Broken Heart
  </h2>
  <p
    className="carousel-author-name"
  >
    Okeremeta - Sep 20, 2018
  </p>
  <p>
    My task is getting bigger and I am equal to the task.
      Music gives a soul to the universe, wings to the mind,
     ...
  </p>
  <button
    className="btn read-more"
    type="button"
  >
    Read more
  </button>
</div>
`;
